src/api/BrowserTypes.ts

```diff
@@ -0,0 +1,5863 @@
+ * The message payload, containing data pertaining to this connection step.
+ */
+export interface WebConnectionProtocol5ValidateAppIdentitySuccessResponsePayload {
+    /**
+     * The appId that the app's identity was validated against
+     */
+    appId: string;
+    /**
+     * Implementation metadata for the Desktop Agent, which includes an appMetadata element
+     * containing a copy of the app's own metadata
+     */
+    implementationMetadata: ImplementationMetadata;
+    /**
+     * The instance Id granted to the application by the Desktop Agent.
+     */
+    instanceId: string;
+    /**
+     * Instance UUID associated with the instanceId granted, which may be used to retrieve the
+     * same instance Id if the app is reloaded or navigates.
+     */
+    instanceUuid: string;
+}
+
+/**
+ * Identifies the type of the connection step message.
+ */
+
+/**
+ * Goodbye message to be sent to the Desktop Agent when disconnecting (e.g. when closing the
+ * window or navigating). Desktop Agents should close the MessagePort after receiving this
+ * message, but retain instance details in case the application reconnects (e.g. after a
+ * navigation event).
+ *
+ * A message used during the connection flow for an application to a Desktop Agent in a
+ * browser window. Used for messages sent in either direction.
+ */
+export interface WebConnectionProtocol6Goodbye {
+    /**
+     * Metadata for a disconnection step message
+     */
+    meta: WebConnectionProtocol6GoodbyeMeta;
+    /**
+     * Identifies the type of the connection step message.
+     */
+    type: "WCP6Goodbye";
+}
+
+/**
+ * Metadata for a disconnection step message
+ *
+ * Metadata for this connection step message
+ */
+export interface WebConnectionProtocol6GoodbyeMeta {
+    timestamp: Date;
+}
+
+/**
+ * Identifies the type of the connection step message.
+ */
+
+/**
+ * A message used during the connection flow for an application to a Desktop Agent in a
+ * browser window. Used for messages sent in either direction.
+ */
+export interface WebConnectionProtocolMessage {
+    meta: WebConnectionProtocolMessageMeta;
+    /**
+     * The message payload, containing data pertaining to this connection step.
+     */
+    payload?: { [key: string]: any };
+    /**
+     * Identifies the type of the connection step message.
+     */
+    type: ConnectionStepMessageType;
+}
+
+/**
+ * Metadata for a disconnection step message
+ *
+ * Metadata for this connection step message
+ */
+export interface WebConnectionProtocolMessageMeta {
+    timestamp:              Date;
+    connectionAttemptUuid?: string;
+}
+
+/**
+ * Identifies the type of the connection step message.
+ */
+export type ConnectionStepMessageType = "WCP1Hello" | "WCP2LoadUrl" | "WCP3Handshake" | "WCP4ValidateAppIdentity" | "WCP5ValidateAppIdentityFailedResponse" | "WCP5ValidateAppIdentityResponse" | "WCP6Goodbye";
+
+// Converts JSON strings to/from your types
+// and asserts the results of JSON.parse at runtime
```

Comment:
```suggestion
// and asserts the results of JSON.parse at runtime.
```

Reply:
Won't fix - that comment is generated by quicktype